core/shared/src/main/scala/cats/effect/IO.scala

```diff
@@ -2334,12 +2334,9 @@ object IO extends IOCompanionPlatform with IOLowPriorityImplicits with TuplePara
 }
 
 private object SyncStep {
-  private val MaxSteps = 512
-  def interpret[G[+_], B](io: IO[B], limit: Int)(
-      implicit G: Sync[G]): G[Either[IO[B], (B, Int)]] = {
-    interpret(io, limit, MaxSteps)
-  }
-  def interpret[G[+_], B](io: IO[B], limit: Int, stepsUntilDefer: Int = MaxSteps)(
+  val MaxSteps = 512 // 512 matches  with trampoline depth in IOFiber for stack safety
```

Comment:
This allows the compiler to inline it
```suggestion
  final val MaxSteps = 512 // 512 matches  with trampoline depth in IOFiber for stack safety
```

Reply:
oh ok thank you, am adding it